libraries/loader/src/loader.js

```diff
@@ -158,9 +158,20 @@ function loader(componentDirectory, virtualComponents = {}) {
       }
       return loaded[target];
     }
-
-    return load(target);
+    return myLoader(target);
   };
 }
 
-module.exports = loader;
+
+
+/*
+  Method to crash the process if error occours
+*/
+function crashOnError(target)
```

Comment:
Ahh, the global assignment to the last created loader is so that this method can access it here.

What might be a prefered solution is to set the `crashOnError` function as a property of the closure which is returned, rather than as a module function.

Reply:
I thought of making a `class` first but I failed when I tested it out so , as this idea was given by bstack . So I thought of doing that way only.

Reply:
There's not really a need to make this a class, and that would likely result in a fairly large change to the the projects which use the loader.

Since a closure is a function which returns a function, and a function is an object which can have properties, the `crashOnError` function could be attached as a property of the function that's returned.

For example:
```javascript
function makeClosure(x) {
  let closure =  function(y) {
    console.log('From closure scope:', x, 'From argument:', y);
  }

  closure.doubler = function(y) {
    console.log('From closure scope:', x * 2, 'From argument:', y * 2);
  }

  return closure;
}

let closure = makeClosure(10);

closure(20);
closure.doubler(20);
```
will print
```bash
$ node closure.js 
From closure scope: 10 From argument: 20
From closure scope: 20 From argument: 40
```